### ISSUE TITLE
Do you have plans to integrate mailing features?

### ISSUE THREAD

**Issue by samwell-ho** [saturneric/GpgFrontend#187]
Do you have plans to integrate mailing features?

I think JMAP is a good choice. Try sending mail in JMAP with Stalwart mail server.

**Comment by saturneric** [comment]
Yes, surely, but the mailing function is too complex and needs lots of work. So, at the first stage, I don't want to add a full-featured email client in GpgFrontend. Instead, I plan to add some tiny functions like verifying email by .emi file or send an email through STMP server. I'm wondering if this functions helpful.

**Comment by samwell-ho** [comment]
Thank you very much for your planning.

I think just the feature of sending mail via SMTP is good enough. If you integrate both receiving and displaying mail features, the software will become complicated.

**Comment by samwell-ho** [comment]
If sending encrypted/signed files via email is allowed, I think files should be in ASSCI amor format instead of binary. Then mail clients can decrypt/verify them.

**Comment by saturneric** [comment]
> If sending encrypted/signed files via email is allowed, I think files should be in ASSCI amor format instead of binary. Then mail clients can decrypt/verify them.

I will follow the standards outlined in [RFC 3156](https://www.rfc-editor.org/rfc/rfc3156.txt) and [RFC 4880](https://www.rfc-editor.org/rfc/rfc4880.txt).

**Comment by saturneric** [comment]
> Thank you very much for your planning.
> 
> I think just the feature of sending mail via SMTP is good enough. If you integrate both receiving and displaying mail features, the software will become complicated.



> Thank you very much for your planning.
> 
> I think just the feature of sending mail via SMTP is good enough. If you integrate both receiving and displaying mail features, the software will become complicated.

I am considering allowing GpgFrontend to edit the plain-text email content, fill in details like sender and recipient addresses, and then encrypt or sign the email. Afterward, the email can be exported as an EML file, which can then be imported into other users' email clients. Do you think this approach could solve some of the issues?

If GpgFrontend were to send the email directly via SMTP, it would generally not be possible to see the sent email in the "Sent" folder. To enable this feature, we would also need to support the IMAP protocol. This would essentially turn GpgFrontend into another email client, and it would need to handle many more aspects, not just focus on GnuPG encryption and signing.

**Comment by saturneric** [comment]
> If GpgFrontend were to send the email directly via SMTP, it would generally not be possible to see the sent email in the "Sent" folder. To enable this feature, we would also need to support the IMAP protocol. This would essentially turn GpgFrontend into another email client, and it would need to handle many more aspects, not just focus on GnuPG encryption and signing.

As v2.1.6 released, these email functions mentioned have added to GpgFrontend. I'll close this issue. If you have any further advice, feel free to open another one.
